rusty_pizza_server/src/money.rs

```diff
@@ -0,0 +1,383 @@
+    type Output = Self;
+
+    fn mul(self, other: u8) -> Self {
+        Self {
+            cents: self.cents * other as u32,
+        }
+    }
+}
+
+impl Mul<Money> for u8 {
+    type Output = Money;
+
+    fn mul(self, other: Money) -> Money {
+        other * self
+    }
+}
+
+impl Mul<u16> for Money {
+    type Output = Self;
+
+    fn mul(self, other: u16) -> Self {
+        Self {
+            cents: self.cents * other as u32,
+        }
+    }
+}
+
+impl Mul<Money> for u16 {
+    type Output = Money;
+
+    fn mul(self, other: Money) -> Money {
+        other * self
+    }
+}
+
+impl Mul<u32> for Money {
+    type Output = Self;
+
+    fn mul(self, other: u32) -> Self {
+        Self {
+            cents: self.cents * other,
+        }
+    }
+}
+
+impl Mul<Money> for u32 {
+    type Output = Money;
+
+    fn mul(self, other: Money) -> Money {
+        other * self
+    }
+}
+
+impl Display for Money {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let euros = self.cents / 100;
+        let cents = self.cents % 100;
+        write!(f, "{},{}€", euros, cents)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fmt::Write;
+
+    #[test]
+    fn money_can_be_created() {
+        // When:
+        let money = Money::new(5, 50);
+
+        // Then:
+        assert_eq!(money, Money { cents: 550 })
+    }
+
+    #[test]
+    fn money_can_be_created_with_alternative_values() {
+        // When:
+        let money = Money::new(7, 20);
+
+        // Then:
+        assert_eq!(money, Money { cents: 720 })
+    }
+
+    #[test]
+    fn money_can_be_added() {
+        // When:
+        let result = Money::new(7, 20) + Money::new(5, 50);
+
+        // Then:
+        assert_eq!(result, Money { cents: 1270 })
+    }
+
+    #[test]
+    fn money_can_be_added_with_different_values() {
+        // When:
+        let result = Money::new(7, 20) + Money::new(5, 55);
+
+        // Then:
+        assert_eq!(result, Money { cents: 1275 })
+    }
+
+    #[test]
+    fn money_can_be_subtracted() {
+        // When:
+        let result = Money::new(7, 20) - Money::new(5, 50);
+
+        // Then:
+        assert_eq!(result, Money { cents: 170 })
+    }
+
+    #[test]
+    fn money_can_be_subtracted_with_different_values() {
+        // When:
+        let result = Money::new(7, 20) - Money::new(5, 55);
+
+        // Then:
+        assert_eq!(result, Money { cents: 165 })
+    }
+
+    #[test]
+    #[should_panic]
+    fn creating_negative_amount_of_money_through_subtraction_panics() {
+        // When:
+        let _ = Money::new(7, 20) - Money::new(7, 40);
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u8() {
+        // When:
+        let result = Money::new(5, 0) * 2u8;
+
+        // Then:
+        assert_eq!(result, Money { cents: 1000 })
+    }
+
+    #[test]
+    fn money_can_be_multiplied_with_an_u8_with_different_values() {
+        // When:
+        let result = Money::new(2, 0) * 3u8;
```

Comment:
If you make a test for different values use something with a cent value in one of the test.

Reply:
Done.